frontend/src/components/molecules/MatchItem/MatchItem.tsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import styled from "styled-components";
+import ListItemAvatar from '@material-ui/core/ListItemAvatar';
+import ListItemText from '@material-ui/core/ListItemText';
+import {Avatar, Card, Typography, withStyles} from '@material-ui/core';
+import {MatchTeam} from "../MatchTeam/MatchTeam";
+import Accordion from '@material-ui/core/Accordion';
+import AccordionDetails from '@material-ui/core/AccordionDetails';
+import AccordionSummary from '@material-ui/core/AccordionSummary';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import {THEME} from "../../atoms/constants/ThemeMUI";
+import AddCircleTwoToneIcon from '@material-ui/icons/AddCircleTwoTone';
+import MatchWinnerDeterminationDialog from "../MatchWinnerDeterminationDialog/MatchWinnerDeterminationDialog";
+
+export type MatchItemProps = {
+    onClickTeam: () => void,
+    matchNumber: number | undefined,
+    level: number | undefined,
+    matchStatus: string | undefined
+    team1: {
+        readonly player1: string | undefined;
+        readonly player2: string | undefined;
+        readonly teamNumber: number | string | undefined;
+        readonly currentPlayerLevel: number | undefined;
+        readonly currentMatchNumber: number | undefined;
+    },
+    team2: {
+        readonly player1: string | undefined;
+        readonly player2: string | undefined;
+        readonly teamNumber: string | undefined;
+        readonly currentPlayerLevel: number | undefined;
+        readonly currentMatchNumber: number | undefined;
+    },
+
+    expanded: string | boolean,
+    handleChangeExpander: (panel: string | boolean) => (event: any, isExpanded: string | boolean) => void,
+};
+
+const TeamHoverButton = styled("div")({
+    "zIndex": 1,
+    boxSizing: "border-box",
+    display: 'inline-flex',
+    position: "absolute",
+    width: "100%",
+    height: "100%",
+    margin: 0,
+    padding: 0,
+    "justify-content": "center",
+    "align-items": "center",
+    fontSize: "65px",
+    backgroundColor: THEME.palette.success.light,
+    opacity: 0,
+    cursor: "pointer",
+    transition: "opacity 1s",
+    '&:hover': {
+        opacity: 0.8,
+    },
+});
+
+const MatchTeamWrapper = styled(Card)({
+    position: "relative",
+    margin: "3px"
+});
+
+const StyledAccordionDetails = styled(AccordionDetails)({
+    display: "flex",
+    flexDirection: "column",
+    backgroundColor: THEME.palette.background.default,
+});
+
+const StyledAvatar = withStyles({
+    root: {
+        color: THEME.palette.getContrastText(THEME.palette.secondary.main),
+        backgroundColor: THEME.palette.secondary.main,
+    },
+})(Avatar);
+
+const MatchItemWrapper = withStyles({
+    root: {
+        width: "100%"
+    }
+})(Card);
```

Comment:
Ja bym wyrzucił te wszystkie style do osobnego pliku. Tak samo w MatchTeam. Tak przynajmniej do tego pory było robione :-)

Reply:
Jak dla mnie ten komponent to szczegół implementacyjny i może zostać nie eksportowany. Ale jak wam wygodnie :)

Reply:
@Szambelan  Pawle, jakiś przyklad, nie do końca rozumiem?